packages/block-editor/src/components/global-styles/background-panel.js

```diff
@@ -679,6 +682,54 @@ export default function BackgroundPanel( {
 	headerLabel = __( 'Background image' ),
 	themeFileURIs,
 } ) {
+	const { globalStyles, _links } = useSelect( ( select ) => {
+		const { getSettings } = select( blockEditorStore );
+		const _settings = getSettings();
+		return {
+			globalStyles: _settings[ globalStylesDataKey ],
+			_links: _settings[ globalStylesLinksDataKey ],
+		};
+	}, [] );
+
+	themeFileURIs = themeFileURIs || _links?.[ 'wp:theme-file' ];
+
+	/*
+	 * Resolve any inherited "ref" pointers.
+	 * Should the block editor need inherited values
+	 * across all controls, this could be abstracted.
+	 */
+	const resolvedInheritedValue = useMemo( () => {
+		const resolvedValues = {
+			background: {},
+		};
+
+		if ( ! inheritedValue?.background ) {
+			return inheritedValue;
+		}
+
+		Object.entries( inheritedValue?.background ).forEach(
+			( [ key, backgroundValue ] ) => {
+				resolvedValues.background[ key ] = getResolvedRefValue(
+					backgroundValue,
+					{
+						styles: globalStyles,
+					}
+				);
+				if (
+					'backgroundImage' === key &&
+					resolvedValues.background[ key ]?.url
+				) {
+					resolvedValues.background[ key ].url =
+						getResolvedThemeFilePath(
+							resolvedValues.background[ key ].url,
+							themeFileURIs
+						);
+				}
+			}
+		);
+		return resolvedValues;
+	}, [ globalStyles, inheritedValue ] );
```

Comment:
Here I'm consolidating all the data and resolution logic. 

That now includes `themeURIs` for relative URIs, which really only affects background images.

It's a first step towards linking globalStyles with all styles controls, at least for one panel. 

Where themeURIs have been passed around as props before, I've removed it.